flink-contrib/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBListState.java

```diff
@@ -158,4 +158,28 @@ public void mergeNamespaces(N target, Collection<N> sources) throws Exception {
 			throw new Exception("Error while merging state in RocksDB", e);
 		}
 	}
+
+	@Override
+	public void update(List<V> values) throws Exception {
+		clear();
+
+		if (values != null && !values.isEmpty()) {
```

Comment:
Is it expected behaviour that adding an empty list will not result in an entry? Maybe that is correct, but we should double check.

Reply:
I believe it's the behavior in rocksdb. Good to have a second eye on it https://github.com/facebook/rocksdb/blob/master/utilities/merge_operators/string_append/stringappend2.cc  since I never wrote c++ before

Reply:
What I meant, is not a RocksDB question, but a question if the list state should return `null` or empty iterator if after the user stored an empty list under a key. That might make a difference and right now, I the caller updates with empty list, the get is identical to a non-existing mapping (`null`).

Reply:
in that case, for all state backends, it returns `null`. I will add a unit test to verify and enforce it.